include/mirage/config.h

```diff
@@ -28,9 +28,13 @@ uint16_t const FP_PQ = 25651;
 uint16_t const FP_P_MUL_Q_MOD_1 = 227;
 // FP_Q_MUL_P_MOD_1 is a multiplier of Q and is 1 module P
 uint16_t const FP_Q_MUL_P_MOD_1 = 25425;
-size_t const MAX_SMEM_SIZE = 96 * 1024;                           // 96 KB
+size_t const MAX_SMEM_SIZE = 96 * 1024;                      // 96 KB
+// Note that we actually save stensors' fingerprints on GPU device memory
+// so MAX_SMEM_FP_SIZE can be larger than MAX_SMEM_SIZE
+size_t const MAX_SMEM_FP_SIZE = 1024 * 1024;                 // 1 MB
 size_t const MAX_DMEM_DATA_SIZE = (size_t)2 * 1024 * 1024 * 1024; // 2 GB
 size_t const MAX_DMEM_FP_SIZE = (size_t)2 * 1024 * 1024 * 1024;   // 2 GB
+size_t const MAX_NUM_THREADBLOCKS_PER_KERNEL = 1024;
```

Comment:
Probably 1024 threadblocks per kernel is too small to fully utilize the GPU...

Reply:
However, on the other side, if we set this to a larger number, the scratchpad memory for fingerprints calculation will be too big...

Reply:
Yeah, it's a trade-off between memory consumption and parallelism. This is also an overhead users have to pay when using mirage to search for mugraphs, so we probably don't want to make it very high. The current overhead is around 1GB for storing fingerprints.